verilog/tools/ls/symbol-table-handler.cc

```diff
@@ -0,0 +1,170 @@
+// Copyright 2021 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "verilog/tools/ls/symbol-table-handler.h"
+
+#include <filesystem>
+
+#include "common/strings/line_column_map.h"
+
+namespace verilog {
+
+static constexpr absl::string_view fileschemeprefix = "file://";
+
+bool LSPUriToPath(absl::string_view uri, absl::string_view path) {
+  if (!absl::StartsWith(uri, fileschemeprefix)) return false;
+  path = uri.substr(fileschemeprefix.size());
```

Comment:
this will modify the local string_view, but will not return (string_view is passed by value).

But before considering a string_view pointer: I'd return it simply; The question is: do you need a boolean telling if the URI was modified ? If not, we could just return the string_view instead.

```
return absl::StartsWith(uri, fileschemeprefix) ? uri.substr(fileschemeprefix.size()) : uri;
```